### PR TITLE
Module tests to master

### PR DESCRIPTION
**Pull request by nkeller1**

### CODE REVIEW

**Review by Will-D-M — COMMENTED** [review]
Final product

File: lib/game.rb [hunk 1]
```diff
@@ -1,18 +1,15 @@
 class Game
 
-  attr_reader :game_id, :season, :type, :date_time, :away_team_id,
-              :home_team_id, :away_goals, :home_goals, :venue,
-              :venue_link
+  attr_reader :game_id, :season, :away_team_id,
```
Comment (madelynrr) [inline lib/game.rb:3]: Great, taking out unnecessary attr reader methods

File: lib/game.rb [hunk 2]
```diff
@@ -1,18 +1,15 @@
...
+              :home_team_id, :away_goals, :home_goals
```
Comment (madelynrr) [inline lib/game.rb:4]: Same, taking out unnecessary pieces.

File: lib/game_collection.rb [hunk 3]
```diff
@@ -30,17 +33,17 @@ def biggest_blowout
 
   def percentage_home_wins
     h_win = @games.count { |game| game.away_goals < game.home_goals }
-    ((h_win * 100.00).to_f / @games.length).round(2)
+    ((h_win).to_f / @games.length).round(2)
```
Comment (madelynrr) [inline lib/game_collection.rb:36]: Change to work in rspec, excellent.

File: lib/game_collection.rb [hunk 4]
```diff
@@ -9,8 +9,11 @@ def initialize(csv_file_path)
   end
 
   def create_games(csv_file_path)
-    csv = CSV.foreach("#{csv_file_path}", headers: true, header_converters: :symbol)
-    csv.map { |row| Game.new(row) }
+    game_array = []
```
Comment (madelynrr) [inline lib/game_collection.rb:12]: Excellent, now not iterating over twice.

File: lib/game_collection.rb [hunk 5]
```diff
@@ -30,17 +33,17 @@ def biggest_blowout
...
   def percentage_home_wins
     h_win = @games.count { |game| game.away_goals < game.home_goals }
-    ((h_win * 100.00).to_f / @games.length).round(2)
+    ((h_win).to_f / @games.length).round(2)
   end
 
   def percentage_visitor_wins
     v_win = @games.count { |game| game.away_goals > game.home_goals }
-    ((v_win * 100.00).to_f / @games.length).round(2)
+    ((v_win).to_f / @games.length).round(2)
```
Comment (madelynrr) [inline lib/game_collection.rb:41]: Change for rspec

File: lib/game_collection.rb [hunk 6]
```diff
@@ -30,17 +33,17 @@ def biggest_blowout
...
   end
 
   def percentage_ties
     ties = @games.count { |game| game.away_goals == game.home_goals }
-    ((ties * 100.00).to_f / @games.length).round(2)
+    ((ties).to_f / @games.length).round(2)
```
Comment (madelynrr) [inline lib/game_collection.rb:46]: Change for rpsec

File: lib/game_team.rb [hunk 7]
```diff
@@ -1,23 +1,11 @@
 class GameTeam
-  attr_reader :game_id, :team_id, :HoA, :result, :settled_in, :head_coach,
-  :goals, :shots, :tackles, :pim, :powerPlayOpportunities, :powerPlayGoals,
-  :faceOffWinPercentage, :giveaways, :takeaways
+  attr_reader :team_id, :HoA, :result, :goals
```
Comment (madelynrr) [inline lib/game_team.rb:2]: Pare down to what we actually use

File: lib/game_team.rb [hunk 8]
```diff
@@ -1,23 +1,11 @@
...
 
   def initialize(game_teams_info)
-    @game_id = game_teams_info[:game_id].to_i
+
     @team_id = game_teams_info[:team_id].to_i
-    @HoA = game_teams_info[:HoA]
+    @HoA = game_teams_info[:hoa]
```
Comment (madelynrr) [inline lib/game_team.rb:7]: Same, leave only what we used

File: lib/game_team_collection.rb [hunk 9]
```diff
@@ -1,23 +1,105 @@
...
   def initialize(csv_file_path)
-    @game_team = create_game_teams(csv_file_path)
+    @game_teams = create_game_teams(csv_file_path)
   end
 
   def create_game_teams(csv_file_path)
-    csv = CSV.foreach("#{csv_file_path}", headers: true, header_converters: :symbol)
-    csv.map do |row|
-      GameTeam.new(row)
+    game_teams = []
```
Comment (madelynrr) [inline lib/game_team_collection.rb:13]: Now not iterating over twice

File: lib/game_team_collection.rb [hunk 10]
```diff
@@ -1,23 +1,105 @@
...
+    game_teams
   end
 
   def count_of_teams
-    team_id_array = @game_team.map { |gt| gt.team_id }
+    team_id_array = @game_teams.map { |gt| gt.team_id }
     team_id_array.uniq.length
   end
 
+  def winningest_team_id
```
Comment (madelynrr) [inline lib/game_team_collection.rb:25]: All new methods from iteration 3

File: lib/helper.rb [hunk 11]
```diff
@@ -0,0 +1,43 @@
+module Helper
```
Comment (madelynrr) [inline lib/helper.rb:1]: Add module for iteration 3

File: lib/team.rb [hunk 12]
```diff
@@ -1,13 +1,8 @@
 class Team
-  attr_reader :team_id, :franchiseId, :teamName, :abbreviation,
-  :Stadium, :link
+  attr_reader :team_id, :teamName
```
Comment (madelynrr) [inline lib/team.rb:2]: Pare down to what we used

File: lib/team.rb [hunk 13]
```diff
@@ -1,13 +1,8 @@
...
 
   def initialize(teams_info)
     @team_id = teams_info[:team_id].to_i
-    @franchiseId = teams_info[:franchiseId].to_i
-    @teamName = teams_info[:teamName]
-    @abbreviation = teams_info[:abbreviation]
-    @Stadium = teams_info[:Stadium]
-    @link = teams_info[:link]
+    @teamName = teams_info[:teamname]
```
Comment (madelynrr) [inline lib/team.rb:6]: Pare down to what we used

File: lib/team_collection.rb [hunk 14]
```diff
@@ -9,8 +9,10 @@ def initialize(csv_file_path)
   end
 
   def create_teams(csv_file_path)
-    csv = CSV.foreach("#{csv_file_path}", headers: true, header_converters: :symbol)
-
-    csv.map { |row| Team.new(row) }
+    team_array = []
```
Comment (madelynrr) [inline lib/team_collection.rb:12]: Great, not iterating through twice now.

File: test/dummy_game_team_data.csv [hunk 15]
```diff
@@ -8,4 +8,21 @@ game_id,team_id,HoA,result,settled_in,head_coach,goals,shots,tackles,pim,powerPl
 2012030236,17,home,WIN,REG,Mike Babcock,3,9,29,12,3,0,61,12,4
 2017030161,30,away,LOSS,REG,Bruce Boudreau,2,5,31,4,1,0,45.6,15,3
 2017030162,30,away,LOSS,REG,Bruce Boudreau,1,4,23,40,4,1,62.3,10,3
-2017030163,52,away,LOSS,REG,Paul Maurice,2,7,26,6,5,1,47.8,7,3
+2012030232,16,home,LOSS,REG,Joel Quenneville,1,5,36,10,2,0,44.6,7,9
```
Comment (madelynrr) [inline test/dummy_game_team_data.csv:11]: Add a dummy CSV file

File: test/fan_collection_test.rb [hunk 16]
```diff
@@ -0,0 +1,39 @@
+# require_relative 'test_helper'
```
Comment (madelynrr) [inline test/fan_collection_test.rb:1]: Test file for the fan collection.

File: test/game_collection_test.rb [hunk 17]
```diff
@@ -25,17 +25,17 @@ def test_it_can_calculate_biggest_blowout
   end
 
   def test_percentage_home_wins
-    assert_equal 33.33, @game_collection.percentage_home_wins
+    assert_equal 0.33, @game_collection.percentage_home_wins
```
Comment (madelynrr) [inline test/game_collection_test.rb:28]: Change so our tests match rspec needs.

File: test/game_team_collection_test.rb [hunk 18]
```diff
@@ -5,11 +5,41 @@
 class GameTeamCollectionTest < Minitest::Test
 
   def setup
-    @game_team_collection= GameTeamCollection.new("./test/test_game_team_data.csv")
+    @game_team_collection= GameTeamCollection.new("./test/dummy_game_team_data.csv")
```
Comment (madelynrr) [inline test/game_team_collection_test.rb:8]: Switch to correct file path for testing.

File: test/game_team_collection_test.rb [hunk 19]
```diff
@@ -5,11 +5,41 @@
...
+  end
+
+  def test_it_exists
+    assert_instance_of GameTeamCollection, @game_team_collection
   end
 
   def test_count_of_teams
-    assert_equal 7, @game_team_collection.count_of_teams
+    assert_equal 12, @game_team_collection.count_of_teams
```
Comment (madelynrr) [inline test/game_team_collection_test.rb:16]: Tests for all game collections.

File: test/game_team_test.rb [hunk 20]
```diff
@@ -0,0 +1,22 @@
+require_relative 'test_helper'
```
Comment (madelynrr) [inline test/game_team_test.rb:1]: Test for game team class

File: test/game_test.rb [hunk 21]
```diff
@@ -0,0 +1,30 @@
+require_relative 'test_helper'
```
Comment (madelynrr) [inline test/game_test.rb:1]: Test for game class

File: test/stat_tracker_test.rb [hunk 22]
```diff
@@ -4,7 +4,7 @@
 
 class StatTrackerTest < Minitest::Test
   def setup
-    @stat_tracker = StatTracker.new("./test/dummy_game_data.csv", "./test/dummy_game_team_data.csv", "./test/dummy_team_data.csv")
+    @stat_tracker = StatTracker.new("./test/dummy_game_data.csv", "./test/dummy_team_data.csv",  "./test/dummy_game_team_data.csv")
```
Comment (madelynrr) [inline test/stat_tracker_test.rb:7]: Put dummy file paths in correct order

File: test/stat_tracker_test.rb [hunk 23]
```diff
@@ -24,17 +24,16 @@ def test_it_can_calculate_biggest_blowout
   end
 
   def test_percentage_home_wins
-    assert_equal 33.33, @stat_tracker.percentage_home_wins
-    #3 wins in test sample
+    assert_equal 0.33, @stat_tracker.percentage_home_wins
```
Comment (madelynrr) [inline test/stat_tracker_test.rb:27]: Change for rspec

File: test/stat_tracker_test.rb [hunk 24]
```diff
@@ -76,4 +75,25 @@ def test_it_can_get_average_goals_per_season
   def test_it_can_return_average_goals_per_game
     assert_equal 3.92, @stat_tracker.average_goals_per_game
   end
+
```
Comment (madelynrr) [inline test/stat_tracker_test.rb:78]: Tests for fan collection.

File: test/team_collection_test.rb [hunk 25]
```diff
@@ -0,0 +1,17 @@
+require_relative 'test_helper'
```
Comment (madelynrr) [inline test/team_collection_test.rb:1]: Test for team collection.

File: test/team_test.rb [hunk 26]
```diff
@@ -0,0 +1,20 @@
+require_relative 'test_helper'
```
Comment (madelynrr) [inline test/team_test.rb:1]: Test for team class

**Review by madelynrr — APPROVED** [review]
Great job, lots of good changes, mainly making things fit for rspec and adding iteration three team.

**Pull request merged**